domain/keyupdater/service/service.go

```diff
@@ -134,35 +164,97 @@ func (s *Service) GetAuthorisedKeysForMachine(
 		)
 	}
 
-	return append(machineKeys, controllerKeys...), nil
+	return append(userKeys, controllerKeys...), nil
 }
 
 // WatchAuthorisedKeysForMachine will watch for authorised key changes for a
 // give machine name. The following errors can be expected:
 // - [github.com/juju/errors.NotValid] if the machine id is not valid.
+// - [machineerrors.MachineNotFound] if no machine exists for the provided name.
 func (s *WatchableService) WatchAuthorisedKeysForMachine(
 	ctx context.Context,
 	machineName coremachine.Name,
-) (watcher.StringsWatcher, error) {
+) (watcher.NotifyWatcher, error) {
 	if err := machineName.Validate(); err != nil {
 		return nil, fmt.Errorf(
 			"cannot watch authorised keys for machine %q: %w", machineName, err,
 		)
 	}
 
-	return s.watcherFactory.NewNamespaceWatcher(
-		"user_public_ssh_key",
+	err := s.st.CheckMachineExists(ctx, machineName)
```

Comment:
Missing nothing, Remember that services are here to answer questions on behalf of the API. The API needs to be able to ask the question for this machine what authorized keys should be on this machine. Implementation details below this is correct.

Now when we need to expand the business logic in the future we have the context correct.

Reply:
Are we going to use the machineName -> machineID to filter out those events from the stream?

Reply:
no we want to know about all event in this context. The machine name at the moment lives in the model database and this watcher is on the controller database.

We take the machine name because that is the context from which we need to answer the question for the caller.